sys/include/xtimer/tick_conversion.h

```diff
@@ -12,9 +13,56 @@
  * @{
  * @file
  * @brief   xtimer tick <-> seconds conversions for different values of XTIMER_HZ
+ *
  * @author  Joakim Nohlgård <joakim.nohlgard@eistec.se>
+ * @author  Josua Arndt <jarndt@ias.rwth-aachen.de>
  */
 
+/*
+ * As tick_conversion.h is included in multiple files ENABLE_DEBUG can not be
+ * used as it gets redefined "[-Werror]"
+ * To use  the debug print enable module fmt in the project makefile
+ *
+ * USEMODULE += fmt
+ * CFLAGS += -DENABLE_DEBUG_TICKS=1
+ * */
+#ifndef ENABLE_DEBUG_TICKS
+#define ENABLE_DEBUG_TICKS   (0)
+#endif
+#if ENABLE_DEBUG_TICKS
+#ifndef MODULE_FMT
+#error "Module fmt is necessary to ENABLE_DEBUG_TICKS set 'USEMODULE += fmt' \
+in the makefile."
+#endif
+#include <stdio.h>
+#include <string.h>
+#include "fmt.h"
+#define DEBUG_PRINT_TICKS_32( CLOCK_HZ, EQUAL, VALUE1, VALUE2, BASE2) ({\
+char buf_1[21];                                                         \
+char buf_2[21];                                                         \
+size_t size = fmt_u32_dec(buf_1, VALUE1);                               \
+buf_1[size] = '\0';                                                     \
+size = fmt_u32_dec(buf_2, VALUE2);                                      \
+buf_2[size] = '\0';                                                     \
+printf("32bit %" PRIu32 " "EQUAL" 1MHz : %s "#VALUE1" = %s "BASE2"\n",  \
```

Comment:
pick either fmt or stdio as output, there is no reason to use PRIu32 for one argument and preformatting with fmt_u32_dec for the other argument.

Reply:
there are `print_xxx` functions in fmt.h which you can use as a convenience instead of managing temporary buffers manually.

Reply:
But there is no real benefit in `print_xxx` as the code then just splits the print in multiple lines.